### PR TITLE
comparable task scheduler handle exception

### PR DESCRIPTION
**Pull request by bchong95**
# comparable task scheduler handle exception

## Description

Handling exception 

## Type of change

Please delete options that are not relevant.

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Closing issues

Put closes #XXXX in your comment to auto-close the issue that your PR fixes (if such).

## Assignee

Please add yourself as the assignee

## Projects

Please add relevant projects so this issue can be properly tracked.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Query/Core/ExecutionComponent/Distinct/DistinctMap.UnorderedDistinctMap.cs [hunk 1]
```diff
@@ -158,49 +159,14 @@ private sealed class UnorderdDistinctMap : DistinctMap
...
-
                 this.utf8Buffer = new byte[UnorderdDistinctMap.UInt128Length];
-                this.numbers = numbers;
-                this.stringsLength4 = stringsLength4;
-                this.stringsLength8 = stringsLength8;
-                this.stringsLength16 = stringsLength16;
-                this.stringsLength16Plus = stringsLength16Plus;
-                this.arrays = arrays;
-                this.objects = objects;
+                this.numbers = numbers ?? throw new ArgumentNullException(nameof(numbers));
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/Core/ExecutionComponent/Distinct/DistinctMap.UnorderedDistinctMap.cs]: Thanks for doing this. It's a huge improvement in readability.

File: Microsoft.Azure.Cosmos/src/Query/Core/ComparableTask/ComparableTaskScheduler.cs [hunk 2]
```diff
@@ -165,6 +156,9 @@ private async Task ExecuteComparableTaskAsync(IComparableTask comparableTask)
                 comparableTask.StartAsync(this.CancellationToken)
                     .ContinueWith((antecendent) =>
                     {
+                        // Observing the exception.
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/Core/ComparableTask/ComparableTaskScheduler.cs]: Should the refactoring and this fix be separate PRs? It's difficult to tell what the fix with all this refactoring.

File: Microsoft.Azure.Cosmos/src/Query/Core/ComparableTask/ComparableTaskScheduler.cs [hunk 3]
```diff
@@ -165,6 +156,9 @@ private async Task ExecuteComparableTaskAsync(IComparableTask comparableTask)
...
+                        Exception exception = antecendent.Exception;
+                        Extensions.TraceException(exception);
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/Core/ComparableTask/ComparableTaskScheduler.cs]: I might have missed it, but did we add any tests for this?

File: Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/FeedManagement/RemainingWorkEstimatorCore.cs [hunk 4]
```diff
@@ -17,6 +17,7 @@ namespace Microsoft.Azure.Cosmos.ChangeFeed.FeedManagement
     using Microsoft.Azure.Cosmos.ChangeFeed.LeaseManagement;
     using Microsoft.Azure.Cosmos.ChangeFeed.Utils;
     using Microsoft.Azure.Cosmos.Core.Trace;
+    using Microsoft.Azure.Cosmos.Query.Core;
```
[the remark this answers is not in the document]
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/FeedManagement/RemainingWorkEstimatorCore.cs]: sorry I branched off another branch that got into master now.

File: Microsoft.Azure.Cosmos/src/Linq/SQLTranslator.cs [hunk 5]
```diff
@@ -53,7 +54,7 @@ internal static class SqlTranslator
             {
                 foreach (KeyValuePair<object, string> keyValuePair in parameters)
                 {
-                    sqlParameters.Add(new Microsoft.Azure.Cosmos.SqlParameter(keyValuePair.Value, keyValuePair.Key));
+                    sqlParameters.Add(new Microsoft.Azure.Cosmos.Query.Core.SqlParameter(keyValuePair.Value, keyValuePair.Key));
                 }
             }
             queryText = query.ToString();
```
[the remark this answers is not in the document]
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Linq/SQLTranslator.cs]: It's also used in Json{Reader, Writer, Navigator} and SqlObjects.

File: Microsoft.Azure.Cosmos/src/Query/Core/ComparableTask/ComparableTaskScheduler.cs (hunk 2 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/Core/ComparableTask/ComparableTaskScheduler.cs]: yes it should

File: Microsoft.Azure.Cosmos/src/Query/Core/ComparableTask/ComparableTaskScheduler.cs (hunk 3 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/Core/ComparableTask/ComparableTaskScheduler.cs]: We don't have any tests plus it would be difficult to write.

**Comment by kirankumarkolli** [comment]
Latest version has only a single file change that too tracing.
Is it expected?
